core/transaction-pool/graph/src/validated_pool.rs

```diff
@@ -219,6 +219,12 @@ impl<B: ChainApi> ValidatedPool<B> {
 				}
 			}
 
+			// if we're rejecting future transactions, then insertion order matters here:
+			// if tx1 depends on tx2, then if tx1 is inserted before tx2, then it goes
+			// to the future queue and gets rejected immediately
+			// => let's temporary stop rejection and clear future queue before return
+			let reject_future_transactions = pool.reject_future_transactions(false);
```

Comment:
To be fair I'd prefer either `pool.with_futures_enabled(FnOnce)` or a guard that resets to previous value on `Drop`.
Just setting a flag and flipping it back in the end is error prone, especially if you accidentally `return` or use `?` at some point .